CONTRIBUTING.md

```diff
@@ -0,0 +1,61 @@
+# Contribution Guidelines
+
+Thank you for your interest in contributing to Unity Robotics! To facilitate your 
+contributions, we've outlined a brief set of guidelines to ensure that your extensions 
+can be easily integrated.
+
+## Communication
+
+First, please read through our
+[code of conduct](CODE_OF_CONDUCT.md),
+as we expect all our contributors to follow it.
+
+Second, before starting on a project that you intend to contribute to any of our
+Unity Robotics packages or tutorials, we **strongly** recommend posting on the repository's
+[Issues page](https://github.com/Unity-Technologies/ROS-TCP-Endpoint/issues) and
+briefly outlining the changes you plan to make. This will enable us to provide
+some context that may be helpful for you. This could range from advice and
+feedback on how to optimally perform your changes or reasons for not doing it.
+
+Lastly, if you're looking for input on what to contribute, feel free to reach
+out to our team directly at [unity-robotics@unity3d.com](mailto:unity-robotics@unity3d.com) 
+and/or browse the GitHub issues.
+
+## Git Branches
+
+The `main` branch corresponds to the most recent stable version of the project. The `dev` branch
+contains changes that are staged to be merged into `main` as the team sees fit.
+
+When contributing to the project, please make sure that your Pull Request (PR)
+does the following:
+
+- Is up-to-date with and targets the `dev` branch
+- Contains a detailed description of the changes performed
+- Has corresponding changes to documentation, unit tests and sample environments (if
+  applicable)
+- Contains a summary of the tests performed to validate your changes
+- Links to issue numbers that the PR resolves (if any)
+
+<!-- ## Continuous Integration (CI)
+
+We run continuous integration on all PRs; all tests must be passing before the PR is merged. -->
```

Comment:
I'm going to be looking into whether or not it's possible to run CI *before* merges in the case where we get a PR from someone not on the code owner's list.  In cases where we can't follow up easily with the submitter if the code breaks `dev` after merge, we really want to be confident that the tests are going to pass before it goes in.

Reply:
Looks like there's no way to change test behavior based on the actual submitter. I think the best we can do is change it to run CI on every PR submission and give code owners the right to over-ride the status check on as as-needed basis.